javagen/src/main/java/com/azure/autorest/mapper/ProxyMethodExampleMapper.java

```diff
@@ -32,7 +36,20 @@ public ProxyMethodExample map(Object xmsExample) {
             }
             String xmsOriginalFile = (String) ((Map<String, Object>) xmsExample).get("x-ms-original-file");
             builder.originalFile(xmsOriginalFile);
+            if (exampleWrapper.getOperationId() != null) {
+                builder.codeSnippetIdentifier(buildCodeSnippetIdentifier(exampleWrapper.getOperationId()));
+            }
         }
         return builder.build();
     }
+
+    private String buildCodeSnippetIdentifier(String operationId) {
+        String[] operationGroupAndId = operationId.split("_");
+        if (operationGroupAndId.length != 2) {
+            return null;
+        }
+        String operationGroup = operationGroupAndId[0];
+        String oprId = operationGroupAndId[1];
+        return String.format("%s.%s.%s", JavaSettings.getInstance().getPackage(), operationGroup.toLowerCase(Locale.ROOT), oprId.toLowerCase(Locale.ROOT));
```

Comment:
It should be operation group + operation + example name

Maybe you can add operationId and example name, then exclude special char.